backends/aoti/runtime/AotiBackend.cpp

```diff
@@ -0,0 +1,561 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/extension/tensor/tensor.h>
+#include <executorch/runtime/backend/interface.h>
+#include <executorch/runtime/core/error.h>
+#include <executorch/runtime/core/evalue.h>
+
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include "cuda_runtime.h"
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+namespace executorch {
+namespace backends {
+namespace aoti {
+
+// Here is where the aoti bouncers are going to be defined.
+// I define the globals aoti generated compiled code calls
+// They can be backed by ET systems
+
+using namespace std;
+
+using executorch::aten::ScalarType;
+using executorch::runtime::ArrayRef;
+using executorch::runtime::Backend;
+using executorch::runtime::BackendExecutionContext;
+using executorch::runtime::BackendInitContext;
+using executorch::runtime::CompileSpec;
+using executorch::runtime::DelegateHandle;
+using executorch::runtime::Error;
+using executorch::runtime::EValue;
+using executorch::runtime::FreeableBuffer;
+using executorch::runtime::MemoryAllocator;
+using executorch::runtime::Result;
+using executorch::runtime::etensor::Tensor;
+using executorch::runtime::Span;
+
+extern "C" {
+using AOTITensorHandle = Tensor*;
+
+// TODO: We should get a proper one
+struct CUDAStreamGuardOpaque;
+using CUDAStreamGuardHandle = CUDAStreamGuardOpaque*;
+
+using AOTIRuntimeError = Error;
+using AOTITorchError = Error;
+
+struct AOTInductorModelContainerOpaque;
+using AOTInductorModelContainerHandle = AOTInductorModelContainerOpaque*;
+using AOTInductorStreamHandle = void*;
+using AOTIProxyExecutorHandle = void*;
+
+using AOTInductorModelContainerCreateWithDeviceFunc = AOTIRuntimeError (*)(
```

Comment:
A nicer way to get these types is to `#include "torch/csrc/inductor/aoti_runtime/interface.h"` and then use `decltype(&AOTInductorModelContainerCreateWithDevice)` etc. Does require you to *not* create a function pointer named `AOTInductorModelContainerCreateWithDevice`, though.